### PR TITLE
Ability to show only polar hydrogens

### PR DESCRIPTION
**Pull request by dsehnal**
# Description

## Actions

- [ ] Added description of changes to the `[Unreleased]` section of `CHANGELOG.md`
- [ ] Updated headers of modified files
- [ ] Added my name to `package.json`'s `contributors`

### CODE REVIEW

**Comment by jpattle** [comment]
Looks good. For our use case we need the option to _only show polar hydrogens_. Would it be possible to add that functionality too?

I think your `hydrogens` option in `OptionsParams` would need `all`, `hide-all`, `hide-polar` and `only-polar`. Then you could add another boolean flag to type params for 'onlyPolarHydrogens'.

The logic in `bond`, `element` and `common` would need to become:
- `ignoreHydrogens` takes precedent over `ignorePolarHydrogens` and `onlyPolarHydrogens`
- `ignorePolarHydrogens` takes precedent over `onlyPolarHydrogens`

**Comment by dsehnal** [comment]
Actually, just hiding polar doesn't make sense I think, my bad here. I will update it to "showPolarOnly".

File: src/mol-plugin-state/manager/structure/component.ts [hunk 1]
```diff
@@ -458,7 +463,11 @@ class StructureComponentManager extends StatefulPluginComponent<StructureCompone
...
 namespace StructureComponentManager {
     export const OptionsParams = {
-        showHydrogens: PD.Boolean(true, { description: 'Toggle display of hydrogen atoms in representations' }),
+        hydrogens: PD.Select(
+            'all',
+            [['all', 'Show All'], ['hide-all', 'Hide All'], ['only-polar', 'Only Polar']] as const,
+            { description: 'Determine display of hydrogen atoms in representations' }
+        ),
         visualQuality: PD.Select('auto', VisualQualityOptions, { description: 'Control the visual/rendering quality of representations' }),
         ignoreLight: PD.Boolean(false, { description: 'Ignore light for stylized rendering of representtions' }),
```
Comment (jpattle) [inline src/mol-plugin-state/manager/structure/component.ts:472]: typo: representtions -> representations

File: src/mol-repr/structure/visual/util/bond.ts [hunk 2]
```diff
@@ -61,30 +64,49 @@ export function makeIntraBondIgnoreTest(structure: Structure, unit: Unit.Atomic,
...
+
+        if ((!!childUnit && !SortedArray.has(childUnit.elements, elements[aI]))) {
+            return true;
+        }
+
+        if (!allBondTypes && ignoreBondType(include, exclude, _flags[edgeIndex])) {
+            return true;
+        }
+
+        if (ignoreHydrogens || onlyPolarHydrogens) {
```
Comment (jpattle) [inline src/mol-repr/structure/visual/util/bond.ts:81]: Reduce nesting by returning early:

```
if (!ignoreHydrogens && !onlyPolarHydrogens) {
  return false;
}

if(isH(atomicNumber, elements[aI])) {
...
```

File: src/mol-repr/structure/visual/util/element.ts [hunk 3]
```diff
@@ -45,14 +47,29 @@ export function makeElementIgnoreTest(structure: Structure, unit: Unit, props: E
...
+            return true;
+        }
+
+        if (traceOnly && !isTrace(unit, element)) {
+            return true;
+        }
+
+        if (isCoarse) return false;
+
+        if ((ignoreHydrogens || onlyPolarHydrogens) && isH(atomicNumber, element)) {
```
Comment (jpattle) [inline src/mol-repr/structure/visual/util/element.ts:63]: Suggestion: Reduce nesting and duplicate if statements by returning early

```
if (!isH(atomicNumber, element)) {
  return false
}

return ignoreHydrogens || (onlyPolarHydrogens && !hasPolarNeighbor(...));
```

**Review by jpattle — APPROVED** [review]
Looks good to me. Reviewed and tested locally with no issues found. Thank you for this 😄

**Pull request merged**